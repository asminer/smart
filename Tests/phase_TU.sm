
dtmc dex1 := {
  state s, t, a;
  init(s:1);
  arcs(
    s:s:2, s:t:1, s:a:1,
    t:s:1,
    a:s:1
  );
  ph int x := phi_TU(potential(in_state(s)), potential(in_state(a)));
  ph real error := phr_TU(potential(in_state(s)), potential(in_state(a)));
  real px := prob_finite(x);
};

print("\n============================================================\n");
print("\ndtmc example 1\n\n");
print("-------------------------------------------------------\n");
print_ph(dex1.x);
print("-------------------------------------------------------\n");
print("\tShould look like:\n");
print("\t0 -> 1;\n"); 
print("\t0 -> 2;\n");
print("-------------------------------------------------------\n");
print("\tShould be 0.5: ", dex1.px, "\n");
print("\n");


dtmc dex2 := {
  state s1, s2, s3, a;
  init(s1:1);
  arcs(
    s1:s2:1, s1:a:1,
    s2:s3:1, 
    s3:s2:1, s3:s3:1
  );
  ph int x := phi_TU(potential(in_state({s1, s2, s3})), potential(in_state(a)));
  ph real error := phr_TU(potential(in_state({s1, s2, s3})), potential(in_state(a)));
  real px := prob_finite(x);
};

print("============================================================\n");
print("\ndtmc example 2\n\n");
print("-------------------------------------------------------\n");
print_ph(dex2.x);
print("-------------------------------------------------------\n");
print("\tShould look like:\n");
print("\t0 -> 1;\n"); 
print("\t0 -> 2;\n");
print("-------------------------------------------------------\n");
print("\tShould be 0.5: ", dex2.px, "\n");
print("\n");


dtmc dex3 := {
  state s1, s2, s3, t, a;
  init(s1:1);
  arcs(
    s1:s2:1, s1:t:1,
    s2:a:1, 
    t:s3:1,
    s3:a:1
  );
  ph int x := phi_TU(potential(in_state({s1, s2, s3})), potential(in_state(a)));
  ph real error := phr_TU(potential(in_state({s1, s2, s3})), potential(in_state(a)));
  real px := prob_finite(x);
};

print("============================================================\n");
print("\ndtmc example 3\n\n");
print("-------------------------------------------------------\n");
print_ph(dex3.x);
print("-------------------------------------------------------\n");
print("\tShould look like:\n");
print("\t0 -> 1;\n"); 
print("\t0 -> 2;\n");
print("\t1 -> 3;\n");
print("-------------------------------------------------------\n");
print("\tShould be 0.5: ", dex3.px, "\n");
print("\n");


dtmc dex4 := {
  state s1, s2, t, a;
  init(s1:1);
  arcs(
    s1:s1:1, s1:s2:3, s1:t:1,
    s2:s2:1, s2:a:3, s2:t:1
  );
  ph int x := phi_TU(potential(in_state({s1, s2})), potential(in_state(a)));
  ph real error := phr_TU(potential(in_state({s1, s2})), potential(in_state(a)));
  real px := prob_finite(x);
};


print("============================================================\n");
print("\ndtmc example 4\n\n");
print("-------------------------------------------------------\n");
print_ph(dex4.x);
print("-------------------------------------------------------\n");
print("\tShould look like:\n");
print("\t0 -> 1;\n"); 
print("\t0 -> 2;\n");
print("\t1 -> 2;\n"); 
print("\t1 -> 3;\n"); 
print("-------------------------------------------------------\n");
print("\tShould be ", 9/16, ": ", dex4.px, "\n");
print("\n");


ctmc cex1 := {
  state s, t, a;
  init(s:1);
  arcs(
    s:s:2, s:t:1, s:a:1,
    t:s:1,
    a:s:1
  );
  ph int error := phi_TU(potential(in_state(s)), potential(in_state(a)));
  ph real x := phr_TU(potential(in_state(s)), potential(in_state(a)));
  real px := prob_finite(x);
};

print("\n============================================================\n");
print("\nctmc example 1\n\n");
print("-------------------------------------------------------\n");
print_ph(cex1.x);
print("-------------------------------------------------------\n");
print("\tShould look like:\n");
print("\t0 -> 1;\n"); 
print("\t0 -> 2;\n");
print("-------------------------------------------------------\n");
print("\tShould be 0.5: ", cex1.px, "\n");
print("\n");


ctmc cex2 := {
  state s1, s2, s3, a;
  init(s1:1);
  arcs(
    s1:s2:1, s1:a:1,
    s2:s3:1, 
    s3:s2:1, s3:s3:1
  );
  ph int error := phi_TU(potential(in_state({s1, s2, s3})), potential(in_state(a)));
  ph real x := phr_TU(potential(in_state({s1, s2, s3})), potential(in_state(a)));
  real px := prob_finite(x);
};

print("============================================================\n");
print("\nctmc example 2\n\n");
print("-------------------------------------------------------\n");
print_ph(cex2.x);
print("-------------------------------------------------------\n");
print("\tShould look like:\n");
print("\t0 -> 1;\n"); 
print("\t0 -> 2;\n");
print("-------------------------------------------------------\n");
print("\tShould be 0.5: ", cex2.px, "\n");
print("\n");


ctmc cex3 := {
  state s1, s2, s3, t, a;
  init(s1:1);
  arcs(
    s1:s2:1, s1:t:1,
    s2:a:1, 
    t:s3:1,
    s3:a:1
  );
  ph int error := phi_TU(potential(in_state({s1, s2, s3})), potential(in_state(a)));
  ph real x := phr_TU(potential(in_state({s1, s2, s3})), potential(in_state(a)));
  real px := prob_finite(x);
};

print("============================================================\n");
print("\nctmc example 3\n\n");
print("-------------------------------------------------------\n");
print_ph(cex3.x);
print("-------------------------------------------------------\n");
print("\tShould look like:\n");
print("\t0 -> 1;\n"); 
print("\t0 -> 2;\n");
print("\t1 -> 3;\n");
print("-------------------------------------------------------\n");
print("\tShould be 0.5: ", cex2.px, "\n");
print("\n");


ctmc cex4 := {
  state s1, s2, t, a;
  init(s1:1);
  arcs(
    s1:s1:1, s1:s2:3, s1:t:1,
    s2:s2:1, s2:a:3, s2:t:1
  );
  ph int error := phi_TU(potential(in_state({s1, s2})), potential(in_state(a)));
  ph real x := phr_TU(potential(in_state({s1, s2})), potential(in_state(a)));
  real px := prob_finite(x);
};


print("============================================================\n");
print("\nctmc example 4\n\n");
print("-------------------------------------------------------\n");
print_ph(cex4.x);
print("-------------------------------------------------------\n");
print("\tShould look like:\n");
print("\t0 -> 1;\n"); 
print("\t0 -> 2;\n");
print("\t1 -> 2;\n"); 
print("\t1 -> 3;\n"); 
print("-------------------------------------------------------\n");
print("\tShould be ", 9/16, ": ", cex4.px, "\n");
print("\n");


print("============================================================\n");
print("\nerrors:\n\n");
print("-------------------------------------------------------\n");
print_ph(dex1.error);
print_ph(dex2.error);
print_ph(dex3.error);
print_ph(dex4.error);
print_ph(cex1.error);
print_ph(cex2.error);
print_ph(cex3.error);
print_ph(cex4.error);
