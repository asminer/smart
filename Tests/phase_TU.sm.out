
============================================================

dtmc example 1

-------------------------------------------------------
Information for phase type tta(dex1, ...)
Discrete phase type, state has dimension 1
Reachable states:
State 0: 0
State 1: t
State 2: a
3 states total
Accepting state index: 2
Trap state index: 1
Initial distribution:
    (0:1)
Markov chain:
From state 0:
	To state 0 with weight 0.5
	To state 1 with weight 0.25
	To state 2 with weight 0.25
From state 1:
	To state 1 with weight 1
From state 2:
	To state 2 with weight 1
5 edges total
End of information for phase type

-------------------------------------------------------
	Should look like:
	0 -> 1;
	0 -> 2;
-------------------------------------------------------
	Should be 0.5: 0.5

============================================================

dtmc example 2

-------------------------------------------------------
Information for phase type tta(dex2, ...)
Discrete phase type, state has dimension 1
Reachable states:
State 0: 0
State 1: a
State 2: t
3 states total
Accepting state index: 1
Trap state index: 2
Initial distribution:
    (0:1)
Markov chain:
From state 0:
	To state 1 with weight 0.5
	To state 2 with weight 0.5
From state 1:
	To state 1 with weight 1
From state 2:
	To state 2 with weight 1
4 edges total
End of information for phase type

-------------------------------------------------------
	Should look like:
	0 -> 1;
	0 -> 2;
-------------------------------------------------------
	Should be 0.5: 0.5

============================================================

dtmc example 3

-------------------------------------------------------
Information for phase type tta(dex3, ...)
Discrete phase type, state has dimension 1
Reachable states:
State 0: 0
State 1: 1
State 2: t
State 3: a
4 states total
Accepting state index: 3
Trap state index: 2
Initial distribution:
    (0:1)
Markov chain:
From state 0:
	To state 1 with weight 0.5
	To state 2 with weight 0.5
From state 1:
	To state 3 with weight 1
From state 2:
	To state 2 with weight 1
From state 3:
	To state 3 with weight 1
5 edges total
End of information for phase type

-------------------------------------------------------
	Should look like:
	0 -> 1;
	0 -> 2;
	1 -> 3;
-------------------------------------------------------
	Should be 0.5: 0.5

============================================================

dtmc example 4

-------------------------------------------------------
Information for phase type tta(dex4, ...)
Discrete phase type, state has dimension 1
Reachable states:
State 0: 0
State 1: 1
State 2: t
State 3: a
4 states total
Accepting state index: 3
Trap state index: 2
Initial distribution:
    (0:1)
Markov chain:
From state 0:
	To state 0 with weight 0.2
	To state 1 with weight 0.6
	To state 2 with weight 0.2
From state 1:
	To state 1 with weight 0.2
	To state 2 with weight 0.2
	To state 3 with weight 0.6
From state 2:
	To state 2 with weight 1
From state 3:
	To state 3 with weight 1
8 edges total
End of information for phase type

-------------------------------------------------------
	Should look like:
	0 -> 1;
	0 -> 2;
	1 -> 2;
	1 -> 3;
-------------------------------------------------------
	Should be 0.5625: 0.5625


============================================================

ctmc example 1

-------------------------------------------------------
Information for phase type tta(cex1, ...)
Continuous phase type, state has dimension 1
Reachable states:
State 0: 0
State 1: t
State 2: a
3 states total
Accepting state index: 2
Trap state index: 1
Initial distribution:
    (0:1)
Markov chain:
From state 0:
	To state 1 with weight 1
	To state 2 with weight 1
From state 1:
From state 2:
2 edges total
End of information for phase type

-------------------------------------------------------
	Should look like:
	0 -> 1;
	0 -> 2;
-------------------------------------------------------
	Should be 0.5: 0.5

============================================================

ctmc example 2

-------------------------------------------------------
Information for phase type tta(cex2, ...)
Continuous phase type, state has dimension 1
Reachable states:
State 0: 0
State 1: a
State 2: t
3 states total
Accepting state index: 1
Trap state index: 2
Initial distribution:
    (0:1)
Markov chain:
From state 0:
	To state 1 with weight 1
	To state 2 with weight 1
From state 1:
From state 2:
2 edges total
End of information for phase type

-------------------------------------------------------
	Should look like:
	0 -> 1;
	0 -> 2;
-------------------------------------------------------
	Should be 0.5: 0.5

============================================================

ctmc example 3

-------------------------------------------------------
Information for phase type tta(cex3, ...)
Continuous phase type, state has dimension 1
Reachable states:
State 0: 0
State 1: 1
State 2: t
State 3: a
4 states total
Accepting state index: 3
Trap state index: 2
Initial distribution:
    (0:1)
Markov chain:
From state 0:
	To state 1 with weight 1
	To state 2 with weight 1
From state 1:
	To state 3 with weight 1
From state 2:
From state 3:
3 edges total
End of information for phase type

-------------------------------------------------------
	Should look like:
	0 -> 1;
	0 -> 2;
	1 -> 3;
-------------------------------------------------------
	Should be 0.5: 0.5

============================================================

ctmc example 4

-------------------------------------------------------
Information for phase type tta(cex4, ...)
Continuous phase type, state has dimension 1
Reachable states:
State 0: 0
State 1: 1
State 2: t
State 3: a
4 states total
Accepting state index: 3
Trap state index: 2
Initial distribution:
    (0:1)
Markov chain:
From state 0:
	To state 1 with weight 3
	To state 2 with weight 1
From state 1:
	To state 2 with weight 1
	To state 3 with weight 3
From state 2:
From state 3:
4 edges total
End of information for phase type

-------------------------------------------------------
	Should look like:
	0 -> 1;
	0 -> 2;
	1 -> 2;
	1 -> 3;
-------------------------------------------------------
	Should be 0.5625: 0.5625

============================================================

errors:

-------------------------------------------------------
ERROR in file phase_TU.sm near line 11:
    Underlying distribution is ph int instead of ph real
null phase type
ERROR in file phase_TU.sm near line 37:
    Underlying distribution is ph int instead of ph real
null phase type
ERROR in file phase_TU.sm near line 64:
    Underlying distribution is ph int instead of ph real
null phase type
ERROR in file phase_TU.sm near line 90:
    Underlying distribution is ph int instead of ph real
null phase type
ERROR in file phase_TU.sm near line 118:
    Underlying distribution is ph real instead of ph int
null phase type
ERROR in file phase_TU.sm near line 144:
    Underlying distribution is ph real instead of ph int
null phase type
ERROR in file phase_TU.sm near line 171:
    Underlying distribution is ph real instead of ph int
null phase type
ERROR in file phase_TU.sm near line 197:
    Underlying distribution is ph real instead of ph int
null phase type
